mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/DeleteMessageListener.java

```diff
@@ -0,0 +1,122 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres;
+
+import java.util.Optional;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.events.Event;
+import org.apache.james.events.EventListener;
+import org.apache.james.events.Group;
+import org.apache.james.mailbox.events.MailboxEvents.Expunged;
+import org.apache.james.mailbox.events.MailboxEvents.MailboxDeletion;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.apache.james.mailbox.postgres.mail.dao.PostgresMailboxDAO;
+import org.apache.james.mailbox.postgres.mail.dao.PostgresMailboxMessageDAO;
+import org.apache.james.mailbox.postgres.mail.dao.PostgresMessageDAO;
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class DeleteMessageListener implements EventListener.ReactiveGroupEventListener {
+    private static final Optional<PostgresMailboxId> ALL_MAILBOXES = Optional.empty();
+
+    public static class DeleteMessageListenerGroup extends Group {
+
+    }
+
+    private final PostgresMailboxDAO postgresMailboxDAO;
+    private final PostgresMessageDAO postgresMessageDAO;
+    private final PostgresMailboxMessageDAO postgresMailboxMessageDAO;
+    private final BlobStore blobStore;
+
+    public DeleteMessageListener(PostgresMailboxDAO postgresMailboxDAO, PostgresMessageDAO postgresMessageDAO,
```

Comment:
No injection?

Reply:
No injection needed. I used constructor directly.

Reply:
Ok but what Rene means is that you need to bind it into the Pg guice mailbox module.

Reply:
Yes you gonna have issues later :)

Reply:
injected